### DISCUSSION TITLE
How do I correctly map a "Execute Query" API response to an existing (form) field?

### DISCUSSION THREAD

**Discussion by aMacDawg** [Budibase/budibase#15012] [Help] [answered]
**Expected outcome**
As a Budibase user, I wanted to quickly provide an "API sandbox", where public users can input some data in my "long form text" field, click a button which executes a `Query Action`, and view the response from the API request in a designated `json` field

**Actual outcome**
When I click my form's button, there is no output from the request displayed in the `json` field, and I'm not sure how to debug nor what the _actually correct_ syntax should be

**Implementation Details**
you can see the outline of the components, as well as the `binding` for the `Execute Query` action assigned to the `button` component childed to the form
![Screenshot 2024-11-15 at 12 19 02 AM](https://github.com/user-attachments/assets/0479d9fb-d077-4a7a-8957-7e75e51e6374)

I'm not clear if I'm even setting up the JSON body correctly, so I'm sharing an expanded view so you can see the binding configuration and how I _expect (hope?)_ that variable value will be merged into the value at runtime
![Screenshot 2024-11-15 at 12 19 16 AM](https://github.com/user-attachments/assets/7c5c4f16-a946-44de-8331-41edbfdf19a6)

And finally, the `json` field that I wish the request body to be written to -- from the documentation I could find, the instructions say to use a binding like `{{ request }}` but this... doesn't seem right, so I improvised something which seems inline with what Budibase 'expects' but that also didn't work
![Screenshot 2024-11-15 at 12 19 32 AM](https://github.com/user-attachments/assets/7886aa67-a469-43d8-bf16-9eb8a053ccbb)

**_Notes_**
- I wasn't sure how to implement a form that I wanted to just handle as a "pass through" for API requests to my own API server
- so I used a JSON schema (definitions below) for the form just in case and also to unblock the form setup (which required child components to reference form fields which... didn't exist so I defined my own I guess?)
```json
{
  "properties": {
    "essay": {
      "type": "string",
      "title": "Essay",
      "description": "Essay field for compression",
      "required": true
    },
    "responseField": {
      "type": "string",
      "title": "Request Data",
      "description": "Json field for request data response ",
      "required": false
    }
  }
}
```

**Comment by aMacDawg** [comment]
I'd still appreciate some help on this fwiw!

**Comment by avivkri** [comment, marked as answer]
Hello @aMacDawg,

You can add another 'Update Field Value' action after your 'Execute Query' action and refer the result from previous action like so - {{ Action X.Query Result }} or {{ Action X.Query Result.data.[0] }} where X is the action step number seen in UI (1. or 2. and so on)

Here you will find more documentation on these.
https://docs.budibase.com/docs/form-actions#update-field-value
https://docs.budibase.com/docs/chaining-actions#action-referencing

**Reply by aMacDawg** [reply]
Thanks! Now I'm trying to write this value into a `JSON field` on the page form, however doing `{{  parseJSON   Action 1.Query result.data  }}` in the `Set Field Update` action returns `"{{   parseJSON   actions.0.result.data  }}"` when I attempt to run this in production -- removing the `parseJSON` function writes the expected API's response stringified object however

**Reply by aMacDawg** [reply]
Interestingly, I was able to solve this problem with the `literal` syntax instead, as in: `{{   literal  Action 1.Query result.data  }}` and that set the field value as the expected JSON type value

**Comment by mikesealey** [comment]
Hey @aMacDawg 
You've got an action set to execute the query, but I can't see that you're _doing anything_ with that query.

I've got an MS_SQL server set up with the rows from the BB sample data, and a query set to find all rows where first name is like "Suzy".

You can then go one of two ways here - you could use an "Update State" action with the result of that query, and then use State to pass your query response wherever you need it.
![image](https://github.com/user-attachments/assets/a5e2ee18-e193-4ff5-bb85-d898858846bf)

Or you can use the Update Field Value to add it to your form
![image](https://github.com/user-attachments/assets/a8649a2e-d2d4-4b7a-b47d-5ac5790e0beb)
